esx_service/vmdk_ops.py

```diff
@@ -660,11 +660,16 @@ def listVMDK(tenant):
             for x in vmdks]
 
 
-# Return VM managed object, reconnect if needed. Throws if connection fails twice.
-# returns None if the uuid is not found
-def findVmByUuid(vm_uuid):
+
+def findVmByUuid(vm_uuid, is_vc_uuid=False):
+    """
+    Find VM by vm_uuid.
+    is_vc_uuid should be true if vm_uuid is vc uuid, else it should be false.
+    Return VM managed object, reconnect if needed. Throws if connection fails twice.
+    Returns None if the uuid is not found
+    """
     si = get_si()
-    vm = si.content.searchIndex.FindByUuid(None, vm_uuid, True, False)
+    vm = si.content.searchIndex.FindByUuid(None, vm_uuid, True, is_vc_uuid)
```

Comment:
well, bios uuid is retained and duplicated in many situations (e.g. backup / recover) so they are not unique. 
VC also generates UUID clashes between different VCs, that's why VCD implements it's own scheme for uniqueness. 
So while in theory I see your point, in practice there are many shortcuts that damage uniqueness.  
With all that said, I 'd prefer to look for bios UUIDs in BIOS uuids namespace and for VC uuid in VC uuid namespace